### ISSUE TITLE
Feature: To support mix of search strategy

### ISSUE THREAD

**Issue by amedvedjev** [appium/java-client#565]
## Description

    @AndroidFindBy(id = "action_bar")
    @HowToUseLocators(androidAutomation = ALL_POSSIBLE)
    @AndroidFindBy(className = "android.widget.ImageButton")
    @AndroidFindBy(className = "android.widget.ImageView")

it does work now for "ImageView" correctly but for "ImageButton" not! For "ImageButton" it get "action_bar" instead.

**Comment by TikhomirovSergey** [comment]
@amedvedjev It seems it is not the problem of java_client by itself. But it is possible that it is related to 
[ByAll](https://github.com/SeleniumHQ/selenium/blob/master/java/client/src/org/openqa/selenium/support/pagefactory/ByAll.java). 

What will happen if you change annotation sequence like

```java
    @HowToUseLocators(androidAutomation = ALL_POSSIBLE)
    @AndroidFindBy(className = "android.widget.ImageView")
    @AndroidFindBy(className = "android.widget.ImageButton")
    @AndroidFindBy(id = "action_bar")
```

**Comment by amedvedjev** [comment]
what i need is:
1) first search by some element -> @AndroidFindBy(id = "action_bar")
2) then inside search by any element ->
    @AndroidFindBy(className = "android.widget.ImageView")
    @AndroidFindBy(className = "android.widget.ImageButton")

another possibilities are also welcome:
1) search by any -> then inside by any again

**Comment by TikhomirovSergey** [comment]
Ok. I will think about the design. It would be the interesting feature

**Comment by amedvedjev** [comment]
Thanks Sergey! also don't forget about CHAIN possibility.

**Comment by TikhomirovSergey** [comment]
The fix was merged at #646. It is going to be available soon.
